libs/subliminal_patch/providers/whisperai.py

```diff
@@ -0,0 +1,108 @@
+
+        self.video = video
+
+    @property
+    def id(self):
+        return self.video.original_name
+
+    def get_matches(self, video):
+        matches = set()
+
+        if isinstance(video, Episode):
+            matches.update(["series", "season", "episode"])
+        elif isinstance(video, Movie):
+            matches.update(["title"])
+
+        return matches
+
+
+class WhisperAIProvider(Provider):
+    '''Whisper AI Provider.'''
+
+    # These are all the languages Whisper supports.
+    # from whisper.tokenizer import LANGUAGES
+    whisper_languages = ['eng', 'chi', 'ger', 'spa', 'rus', 'kor', 'fre', 'jpn', 'por', 'tur', 'pol', 'cat', 'dut', 'ara', 'swe', 'ita', 'ind', 'hin', 'fin', 'vie', 'heb', 'ukr', 'gre', 'may', 'cze', 'rum', 'dan', 'hun', 'tam', 'nor', 'tha', 'urd', 'hrv', 'bul', 'lit', 'lat', 'mao', 'mal', 'wel', 'slo', 'tel', 'per', 'lav', 'ben', 'srp', 'aze', 'slv', 'kan', 'est', 'mac', 'bre', 'baq', 'ice', 'arm', 'nep', 'mon', 'bos', 'kaz', 'alb', 'swa', 'glg', 'mar', 'pan', 'sin', 'khm', 'sna', 'yor', 'som', 'afr', 'oci', 'geo', 'bel', 'tgk', 'snd', 'guj', 'amh', 'yid', 'lao', 'uzb', 'fao', 'hat', 'pus', 'tuk', 'nno', 'mlt', 'san', 'ltz', 'bur', 'tib', 'tgl', 'mlg', 'asm', 'tat', 'haw', 'lin', 'hau', 'bak', 'jav', 'sun']
+
+    languages = {
+        Language.fromalpha3b(lang) for lang in whisper_languages
+    }
+    languages.update(set(Language.rebuild(lang, hi=True) for lang in languages))
+    languages.update(set(Language.rebuild(lang, forced=True) for lang in languages))
+
+    video_types = (Episode, Movie)
+
+    def __init__(self, endpoint=None, timeout=None):
+        if not endpoint:
+            raise ConfigurationError('Whisper Web Service Endpoint must be provided')
+
+        if not timeout:
+            raise ConfigurationError('Whisper Web Service Timeout must be provided')
+
+        self.endpoint = endpoint
+        self.timeout = int(timeout)
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = 'Subliminal/%s' % __short_version__
+
+    def terminate(self):
+        self.session.close()
+
+    def query(self, language, video):
+        return WhisperAISubtitle(language, video)
+
+    def list_subtitles(self, video, languages):
+        subtitles = [self.query(l, video) for l in languages]
+        return [s for s in subtitles if s is not None]
+
+    def download_subtitle(self, subtitle: WhisperAISubtitle):
+        # Invoke Whisper through the API. This may take a long time depending on the file.
+        # TODO: This loads the entire file into memory, find a good way to stream the file in chunks
+
+        logger.debug("Encoding audio stream to WAV with ffmpeg")
+
+        try:
+            # This launches a subprocess to decode audio while down-mixing and resampling as necessary.
+            # Requires the ffmpeg CLI and `ffmpeg-python` package to be installed.
```

Comment:
To avoid depending on a new package for this simple task, this could probably be replaced with a subprocess function (?)

@morpheus65535 what do you think?